container-search/src/main/java/com/yahoo/prelude/fastsearch/VespaBackEndSearcher.java

```diff
@@ -500,11 +504,21 @@ protected int fillHits(Result result, int packetIndex, Packet[] packets, String
                 DocsumPacket docsum = (DocsumPacket) packets[packetIndex];
 
                 packetIndex++;
-                if ( ! fillHit(fastHit, docsum, summaryClass))
+                try {
+                    if (!fillHit(fastHit, docsum, summaryClass))
+                        skippedHits++;
+                } catch (TimeoutException e) {
```

Comment:
Using an exception to propagate this may not be very good in the scenario where someone asks for very many hits, thus causing a timeout, or if the container is already overloaded, causing some timeouts.
I suggest changing the return value of fillHit to be e.g a FillResult object containing 2 fields
boolean success; // otherwise skipped aka not available
Optional<String> error;

Reply:
Agree.